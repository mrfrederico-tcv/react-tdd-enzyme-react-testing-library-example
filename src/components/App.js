import React, { Component } from 'react'
import { Button } from 'react-bootstrap'
import styled from 'styled-components'

import Gift from './Gift'

const Root = styled.div``
const Title = styled.h4``
const GiftContainer = styled.div``

class App extends Component {
  state = {
    gifts: [],
  }

  handleAddGift = () => {
    const { gifts } = this.state
    const ids = gifts.map(gift => gift.id)
    const maxId = ids.length > 0 ? Math.max(...ids) : 0

    gifts.push({ id: maxId + 1 })

    this.setState({ gifts })
  }

  handleRemoveGift = id => {
    const { gifts } = this.state

    const giftsAux = gifts.filter(gift => gift.id !== id)

    this.setState({ gifts: giftsAux })
  }

  render() {
    const { gifts } = this.state

    return (
      <Root>
        <Title>Gift Giver</Title>
        <GiftContainer testId="GiftContainer">
          {gifts.map(gift => (
            <Gift
              testId="Gift"
              key={gift.id}
              gift={gift}
              onRemoveGift={this.handleRemoveGift}
            />
          ))}
        </GiftContainer>
        <Button testId="Button" onClick={this.handleAddGift}>
          Add Gift
        </Button>
      </Root>
    )
  }
}

export default App
