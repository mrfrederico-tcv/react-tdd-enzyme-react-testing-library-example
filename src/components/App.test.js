import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe('App', () => {
  const app = shallow(<App />)
  const COINDESK_LINK = 'https://www.coindesk.com/price'

  it('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains a connected wallet component', () => {
    expect(app.find('Connect(Wallet)').exists()).toBe(true)
  })

  it('contains a connected Loot component', () => {
    expect(app.find('Connect(Loot)').exists()).toBe(true)
  })

  it('contains a link to the coindesk price page', () => {
    expect(app.find('[testID="CoindeskLink"]').props().href).toBe(COINDESK_LINK)
  })
})
