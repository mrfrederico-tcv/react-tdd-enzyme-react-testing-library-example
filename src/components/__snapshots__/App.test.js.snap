// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<styled.div>
  <styled.h4>
    Loot Check
  </styled.h4>
  <styled.hr />
  <Connect(Wallet) />
  <styled.hr />
  <Connect(Loot) />
  <styled.div>
    Powered by
     
    <styled.a
      href="https://www.coindesk.com/price"
      target="_blank"
      testID="CoindeskLink"
    >
      Coindesk
    </styled.a>
  </styled.div>
</styled.div>
`;
